internal/resolution/variablesources/bundles_and_dependencies_test.go

```diff
@@ -166,34 +255,57 @@ var _ = Describe("BundlesAndDepsVariableSource", func() {
 		// will not appear in this list as we are not currently taking Required GVKs into account
-		Expect(bundle2.Dependencies()).To(WithTransform(CollectDeppyEntities, Equal([]*input.Entity{
-			input.NewEntity("bundle-5", map[string]string{
-				property.TypePackage: `{"packageName": "some-package", "version": "1.5.0"}`,
-				property.TypeChannel: `{"channelName":"stable","priority":0}`,
-			}),
-			input.NewEntity("bundle-4", map[string]string{
-				property.TypePackage: `{"packageName": "some-package", "version": "1.0.0"}`,
-				property.TypeChannel: `{"channelName":"stable","priority":0}`,
-			}),
-		})))
+		Expect(bundle2.Dependencies()).To(HaveLen(2))
+		Expect(bundle2.Dependencies()[0].Name).To(Equal("bundle-5"))
+		Expect(bundle2.Dependencies()[1].Name).To(Equal("bundle-4"))
 	})
 
 	It("should return error if dependencies not found", func() {
-		mockEntitySource = input.NewCacheQuerier(map[deppy.Identifier]input.Entity{})
-		_, err := bdvs.GetVariables(context.TODO(), mockEntitySource)
+		emptyCatalogClient := testutil.NewFakeCatalogClient(make([]*catalogmetadata.Bundle, 0))
+
+		bdvs = variablesources.NewBundlesAndDepsVariableSource(
+			&emptyCatalogClient,
+			&MockRequiredPackageSource{
+				ResultSet: []deppy.Variable{
+					// must match data in fakeCatalogClient
+					olmvariables.NewRequiredPackageVariable("test-package", []*catalogmetadata.Bundle{
+						{Bundle: declcfg.Bundle{
+							Name:    "bundle-2",
+							Package: "test-package",
+							Properties: []property.Property{
+								{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "2.0.0"}`)},
+								{Type: property.TypeGVKRequired, Value: json.RawMessage(`[{"group":"foo.io","kind":"Foo","version":"v1"}]`)},
+								{Type: property.TypePackageRequired, Value: json.RawMessage(`[{"packageName": "some-package", "versionRange": ">=1.0.0 <2.0.0"}]`)},
+							}},
+							InChannels: []*catalogmetadata.Channel{{Channel: declcfg.Channel{Name: "stable"}}},
+						},
+						{Bundle: declcfg.Bundle{
+							Name:    "bundle-1",
+							Package: "test-package",
+							Properties: []property.Property{
+								{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "1.0.0"}`)},
+								{Type: property.TypeGVKRequired, Value: json.RawMessage(`[{"group":"foo.io","kind":"Foo","version":"v1"}]`)},
+							}},
+							InChannels: []*catalogmetadata.Channel{{Channel: declcfg.Channel{Name: "stable"}}},
+						},
+					}),
+				},
+			},
+		)
+		_, err := bdvs.GetVariables(context.TODO())
 		Expect(err).To(HaveOccurred())
-		Expect(err).To(MatchError("could not determine dependencies for entity with id 'bundle-2': could not find package dependencies for bundle 'bundle-2'"))
+		Expect(err.Error()).To(ContainSubstring("could not determine dependencies for bundle with id '-test-package-stable-bundle-2': could not find package dependencies for bundle 'bundle-2'"))
```

Comment:
Nit

> id '-test-package-stable-bundle-2'

This starts with `-` because `CatalogName` is missing in the fake bundle.

I'll update.